packages/bpk-component-map/src/BpkMapMarker.js

```diff
@@ -55,6 +56,12 @@ const BpkMapMarker = (props: Props) => {
     large,
     onClick,
     type,
+    buttonProps: {
+      type: buttonPropsType,
+      className: buttonPropsClassName,
+      onClick: buttonPropsOnClick,
+      ...restButton
```

Comment:
Following the approach used in other components (like [BpkDatepicker](https://github.com/Skyscanner/backpack/blob/master/packages/bpk-component-datepicker/src/BpkDatepicker.js#L107) or [BpkBreadcrumbItem](https://github.com/Skyscanner/backpack/blob/master/packages/bpk-component-breadcrumb/src/BpkBreadcrumbItem.js#L52)), `buttonProps` would be send directly to `button`, but I think it makes sense to extract the props that are already set in the component, to avoid overwriting them.

It's just a proposal, so let me know if you think it's best not to do it and follow the same approach from other components :-)

Reply:
I'm having a difficult time picturing it - so if we went with your proposal, what would the proptypes look like?

Reply:
My idea was to keep it just as an object (like the other examples), as in fact I wouldn't expect any of this "props" to be defined in `buttonProps`. This is just a "safety" measure to make sure users don't overwrite the props defined in the component.

But maybe it's not needed to be that "safe" 😅 

I see that Flow has complained, so not sure what's best: trying to make the proptype more complex or simply keep it as an object and trust the users.

What do you think?

Reply:
Yeah I think it might be best to just make it looser and keep it as an object, it'll make for an simpler API and if users overwrite things, it's on them 😅

Reply:
Makes sense, I've changed it to send all props like in the other components 😅